### DISCUSSION TITLE
Installation in Deno fails

### DISCUSSION THREAD

**Discussion by shivam-tripathi** [mdx-js/mdx#1881] [Q&A] [answered]
I am trying to import and run MDX with Deno, but running into some typescript related errors:
```
import mdx from "https://esm.sh/@mdx-js/mdx@1.6.22?deno"
```
on running gives the following issue in Deno:
```
error: TS2502 [ERROR]: 'thisArg' is referenced directly or indirectly in its own type annotation.
    bind<T>(this: T, thisArg: ThisParameterType<T>): OmitThisParameter<T>;
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at asset:///lib.es5.d.ts:352:22

TS2577 [ERROR]: Return type annotation circularly references itself.
    bind<T>(this: T, thisArg: ThisParameterType<T>): OmitThisParameter<T>;
                                                     ~~~~~~~~~~~~~~~~~~~~
    at asset:///lib.es5.d.ts:352:54
```
I found another somewhat matching issue here:
https://github.com/denoland/deno/issues/9073

## Additional information
```
> deno --version
deno 1.17.1 (release, x86_64-apple-darwin)
v8 9.7.106.15
typescript 4.5.2
```

**Comment by wooorm** [comment]
Hey!

Most (non-trivial) things from npm don’t work in deno. You’ll have more luck asking deno / esm.sh folks.
Also, try MDX 2, theoretically that matches better with deno, as its ESM + TS

**Comment by aiotter** [comment, marked as answer]
The latest build of mdx, `2.0.0-rc.2`, fails to be imported via esm.sh, although older `2.0.0-ci.53` works.

```
> import "https://esm.sh/@mdx-js/mdx@2.0.0-rc.2"
Uncaught SyntaxError: The requested module '/v62/micromark@3.0.10/deno/lib/parse.js' does not provide an export named 'parse'
    at async <anonymous>:2:1
```

You can keep an eye on this issue: https://github.com/esm-dev/esm.sh/issues/179

**Reply by aiotter** [reply]
Now the latest v2.0.0 is importable. https://github.com/esm-dev/esm.sh/issues/179#issuecomment-1031550351
